src/entities/duplicate.js

```diff
@@ -340,6 +340,15 @@ async function duplicateEntities(entities, options) {
             api.history.add({
                 name: 'duplicate entities',
                 undo: () => {
+                    // make sure we get the entities that are in the scene
+                    newEntities = newEntities.map((entity) => {
```

Comment:
Why would an entity in the `newEntities` array not be 'in the scene'? All entities in the hierarchy are in the scene, right?

Reply:
When the user deletes the duplicated entity from scene, newEntities here is now referencing an entity that has been removed

Reply:
We could be storing GUIDs instead of entity objects in duplicate but using entity.latest() is a pattern I saw elsewhere in this project